Library/Homebrew/language/node.rb

```diff
@@ -0,0 +1,33 @@
+module Language
+  module Node
+    def self.npm_cache_config
+      "cache=#{HOMEBREW_CACHE}/npm_cache\n"
+    end
+
+    def self.setup_npm_environment
+      # explicitly use our npm and node-gyp executables instead of the user
+      # managed ones in HOMEBREW_PREFIX/lib/node_modules which might be broken
+      ENV.prepend_path "PATH", Formula["node"].opt_libexec/"npm/bin"
+      # explicitly set npm's cache path to HOMEBREW_CACHE/npm_cache to fix
+      # issues caused by overriding $HOME (long build times, high disk usage)
+      # https://github.com/Homebrew/brew/pull/37#issuecomment-208840366
+      npmrc = Pathname.new("#{ENV["HOME"]}/.npmrc")
+      npmrc.write npm_cache_config unless npmrc.exist?
+    end
+
+    def self.std_npm_install_args(libexec)
+      setup_npm_environment
+      # tell npm to not install .brew_home by adding it to the .npmignore file
+      # (or creating a new one if no .npmignore file already exists)
+      open(".npmignore", "a") { |f| f.write( "\n.brew_home\n") }
+      # npm install args for global style module format installed into libexec
+      ["--verbose", "--global", "--prefix=#{libexec}", "."]
+    end
```

Comment:
There should be no situation in which you have to call `std_npm_install_args` multiple times, especially not inside the same node module.

Through there might be a use case (currently not used in core) where calling `local_npm_install_args` multiple times might be needed: A formula which consists of multiple node modules and other stuff. In this case we would end up indeed with duplications in $PATH. I agree that we should guard the complete method and only the `.npmrc` creation against running multiple times.

Also the method names might not be the best describing names. Maybe something like `local_npm_install_config_and_args` would be better, through this would be already quite long for a method name.